lib/javascripts/base_app.js

```diff
@@ -68,6 +68,10 @@ function BaseApp(zafClient, data) {
   let evt = { firstLoad: true };
   resolveHandler(this, 'app.created')();
   resolveHandler(this, 'app.activated')(evt, evt);
+  $(window).unload(() => {
+    resolveHandler(this, 'app.willDestroy')();
+    delete this.events['app.willDestroy']; // make sure this event only fires once
```

Comment:
That's not gonna solve that issue, is it? I think `zafClient.off('app.willDestroy')` is what you're looking for.